pallets/inflation/src/lib.rs

```diff
@@ -322,28 +341,100 @@ pub mod pallet {
             config.collator_reward_per_block + config.treasury_reward_per_block
         }
 
-        /// Recalculates the inflation based on the total issuance & inflation parameters.
+        /// Recalculates the inflation based on the current total issuance & inflation parameters.
         ///
         /// Returns the new inflation configuration.
         pub(crate) fn recalculate_inflation(next_era: EraNumber) -> InflationConfiguration {
+            // Calculate max emission based on the current total issuance.
             let params = InflationParams::<T>::get();
             let total_issuance = T::Currency::total_issuance();
-
-            // 1. Calculate maximum emission over the period before the next recalculation.
             let max_emission = params.max_inflation_rate * total_issuance;
-            let issuance_safety_cap = total_issuance.saturating_add(max_emission);
 
-            // 2. Calculate distribution of max emission between different purposes.
+            let recalculation_era =
+                next_era.saturating_add(T::CycleConfiguration::eras_per_cycle());
+
+            Self::new_config(recalculation_era, max_emission)
+        }
+
+        /// Re-adjust the existing inflation configuration using the current inflation parameters.
+        ///
+        /// It might seem similar to forcing the inflation recalculation, but it's not.
+        /// This function adjusts the existing configuration, respecting the `max_emission` value used to calculate the current inflation config.
+        /// (The 'force' approach uses the current total issuance)
+        ///
+        /// This call should be used in case inflation parameters have changed during the cycle, and the configuration should be adjusted now.
+        pub(crate) fn readjusted_config() -> InflationConfiguration {
+            // 1. First calculate the params needed to derive the `max_emission` value used to calculate the current inflation config.
+            let config = ActiveInflationConfig::<T>::get();
+
+            // This is reuse from `new_config` function.
+            let blocks_per_cycle = Balance::from(T::CycleConfiguration::blocks_per_cycle().max(1));
+            let build_and_earn_eras_per_cycle =
+                Balance::from(T::CycleConfiguration::build_and_earn_eras_per_cycle().max(1));
+            let periods_per_cycle =
+                Balance::from(T::CycleConfiguration::periods_per_cycle().max(1));
```

Comment:
Why not creating a `get_cycle_parameters` helper function for these parameters to make the code drier? Something like:

```rs
let (blocks_per_cycle, build_and_earn_eras_per_cycle, periods_per_cycle) =
    Self::get_cycle_parameters();
```

Reply:
@ipapandinas I somehow assumed there's more logic behind this, but it's simple conversion from one type to another. I don't want to add a helper function for that, seems completely redundant.